From ad0e37485ccd5e39b2602f8cf133983b924e977a Mon Sep 17 00:00:00 2001
From: Rufus Hamade <rufus.hamade@imgtec.com>
Date: Fri, 27 May 2016 18:00:06 +0100
Subject: [PATCH 101/101] dri/pvr: Add support for YV12

Add the necessary data to allow the IMG DDK to understand
YV12 image data.

Change-Id: I0567e08fb2a77041b0397683b3e078c6f5782ace
Signed-off-by: Rufus Hamade <rufus.hamade@imgtec.com>
---
 src/mesa/drivers/dri/pvr/pvrutil.c | 30 ++++++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/src/mesa/drivers/dri/pvr/pvrutil.c b/src/mesa/drivers/dri/pvr/pvrutil.c
index 0395fad..a253831 100644
--- a/src/mesa/drivers/dri/pvr/pvrutil.c
+++ b/src/mesa/drivers/dri/pvr/pvrutil.c
@@ -188,6 +188,36 @@ static const PVRDRIImageFormat g_asFormats[] =
 				.uiHeightShift = 1
 		},
 	},
+#if defined(__DRI_IMAGE_FOURCC_YVU420)
+	{
+		.eIMGPixelFormat = IMG_PIXFMT_YVU420_3PLANE,
+		.iDRIFourCC = __DRI_IMAGE_FOURCC_YVU420,
+		.iDRIFormat = __DRI_IMAGE_FORMAT_NONE,
+		.iDRIComponents = __DRI_IMAGE_COMPONENTS_Y_U_V,
+		.uiNumPlanes = 3,
+		.sPlanes[0] =
+		{
+				.eIMGPixelFormat = IMG_PIXFMT_R8_UNORM,
+				.iDRIFormat = __DRI_IMAGE_FORMAT_R8,
+				.uiWidthShift = 0,
+				.uiHeightShift = 0
+		},
+		.sPlanes[1] =
+		{
+				.eIMGPixelFormat = IMG_PIXFMT_R8_UNORM,
+				.iDRIFormat = __DRI_IMAGE_FORMAT_R8,
+				.uiWidthShift = 1,
+				.uiHeightShift = 1
+		},
+		.sPlanes[2] =
+		{
+				.eIMGPixelFormat = IMG_PIXFMT_R8_UNORM,
+				.iDRIFormat = __DRI_IMAGE_FORMAT_R8,
+				.uiWidthShift = 1,
+				.uiHeightShift = 1
+		},
+	},
+#endif
 };
 
 
-- 
1.9.1

